### PR TITLE
Fix IllegalArgumentException on JDK 1.8.0_60

### PR DESCRIPTION
**Pull request by dbcowboy**
To determine actual types of parameters for class passed in as lambda to
step definition, code uses old Sun ConstantPool to look for dynamic
linking data.  Previously these values were found at index size()-2;
however as of 1.8.0_60 the location appears to have changed to size()-3.
Given that the following indexes appear to have empty/invalid values,
instead iterate downward from end to find first valid value and use for
parameter type determination.

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
Thanks! Can you follow the [code standard](https://github.com/cucumber/cucumber-jvm/blob/master/CONTRIBUTING.md#contributinghacking) please?

**Comment by dbcowboy** [comment]
Thanks for pointing me to the code standards; sorry I didn't have that in place before I checked in.

Let me know if I should re-submit this change with correct formatting.

**Comment by rnaufal** [comment]
I've just downloaded the latest SNAPSHOT build (1.2.5-SNAPSHOT) and the this change isn't included on it yet. When are you planning to release a new snapshot build?

**Comment by aslakhellesoy** [comment]
@rnaufal it will be in a snaphot jar shortly after this PR is merged.

**Comment by rnaufal** [comment]
Thanks @aslakhellesoy. Does it will get merged soon?

**Comment by aslakhellesoy** [comment]
@rnaufal yeah [RSN](https://en.wiktionary.org/wiki/RSN)

**Comment by rnaufal** [comment]
Hi @aslakhellesoy, did you merge this PR? Thanks in advance!

**Comment by aslakhellesoy** [comment]
@rnaufal the PR is open, so no

File: java8/src/main/java/cucumber/runtime/java8/ConstantPoolTypeIntrospector.java [hunk 1]
```diff
@@ -34,7 +34,20 @@
...
-        String[] memberRef = constantPool.getMemberRefInfoAt(constantPool.getSize() - 2);
+        int size = constantPool.getSize();
+        String[] memberRef = null;
+
+        // find last element in constantPool with valid memberRef
+        // - previously always at size-2 index but changed with JDK 1.8.0_60
+        for (int i = size - 1; i > -1; i--) {
+            try {
+                memberRef = constantPool.getMemberRefInfoAt(i);
+                break;
```
Comment (clicman) [inline java8/src/main/java/cucumber/runtime/java8/ConstantPoolTypeIntrospector.java]: Return should be here.

File: java8/src/main/java/cucumber/runtime/java8/ConstantPoolTypeIntrospector.java [hunk 2]
```diff
@@ -34,7 +34,20 @@
...
+            } catch (IllegalArgumentException e) {
+                // eat error; null entry at ConstantPool index?
+            }
+        }
+
         return memberRef[2];
```
Comment (clicman) [inline java8/src/main/java/cucumber/runtime/java8/ConstantPoolTypeIntrospector.java]: Here must be exception in case something goes wrong with for loop.

**Comment by erran-r7** [comment]
A colleague @kprzerwa-r7 and I encountered this without being sure whether it was ever released. For the next person who stumbles across this issue it was fixed in 96d1e85df4289e290bb4c61893be5516d4dcc9ee (which was released in `1.2.5`).

> [Java8] Fix IllegalArgumentException on JDK 1.8.0_60 (#912, #914 Michael Wilkerson)

**Pull request closed without merging**